packages/views/shared/discreetMode/discreetMode.js

```diff
@@ -1,46 +1,43 @@
-import React, { Component } from 'react';
+import React from 'react';
 import PropTypes from 'prop-types';
 import routes from '@screens/router/routes';
 import { selectSearchParamValue } from '@screens/router/searchParams';
 import styles from './discreetMode.css';
 
-class DiscreetMode extends Component {
-  handleBlurOnOtherWalletPage() {
-    const { account, location: { search }, token } = this.props;
+const DiscreetMode = ({
+  children, location, isDiscreetMode, shouldEvaluateForOtherAccounts,
+  addresses, account, token,
+}) => {
+  const handleBlurOnOtherWalletPage = () => {
+    const { search } = location;
     const address = selectSearchParamValue(search, routes.account.searchParam);
     return account.info && address === account.info[token].address;
-  }
+  };
 
-  shouldEnableDiscreetMode() {
-    const {
-      location, isDiscreetMode, shouldEvaluateForOtherAccounts,
-      addresses, account,
-    } = this.props;
+  const shouldEnableDiscreetMode = () => {
     if (!isDiscreetMode) return false;
     if (shouldEvaluateForOtherAccounts) {
       if (location.pathname.includes(routes.account.path)) {
-        return this.handleBlurOnOtherWalletPage();
+        return handleBlurOnOtherWalletPage();
       }
-      const { location: { search } } = this.props;
+      const { search } = location;
```

Comment:
what happend if the location is not an object? also its need default props
```suggestion
      const { search } = location || {};
```